arpav_ppcv/observations_harvester/operations.py

```diff
@@ -0,0 +1,213 @@
+                station_create = models.StationCreate(
+                    code=station_code,
+                    geom=geojson_pydantic.Point(
+                        type="Point",
+                        coordinates=(pt_4326.x, pt_4326.y)
+                    ),
+                    altitude_m=raw_station["altitude"],
+                    name=raw_station["statnm"],
+                    type_=raw_station["stattype"].lower().replace(" ", "_"),
+                )
+                stations_create[station_create.code] = station_create
+    return list(stations_create.values())
+
+
+def refresh_stations(
+        client: httpx.Client,
+        db_session: sqlmodel.Session
+) -> list[models.Station]:
+    to_create = harvest_stations(client, db_session)
+    logger.info(f"About to create {len(to_create)} stations...")
+    created_variables = database.create_many_stations(db_session, to_create)
+    return created_variables
+
+
+def harvest_monthly_measurements(
+        client: httpx.Client,
+        db_session: sqlmodel.Session,
+        station_id: Optional[uuid.UUID] = None,
+        variable_id: Optional[uuid.UUID] = None,
+) -> list[models.MonthlyMeasurementCreate]:
+    if station_id is not None:
+        existing_stations = [database.get_station(db_session, station_id)]
+    else:
+        existing_stations = database.collect_all_stations(db_session)
+    if variable_id is not None:
+        existing_variables = [database.get_variable(db_session, variable_id)]
+    else:
+        existing_variables = database.collect_all_variables(db_session)
+    monthly_measurements_create = []
+    for station_idx, station in enumerate(existing_stations):
+        logger.info(
+            f"Processing station {station.code!r} ({station_idx+1}/"
+            f"{len(existing_stations)})..."
+        )
+        for var_idx, variable in enumerate(existing_variables):
+            logger.info(
+                f"\tProcessing variable {variable.name!r} ({var_idx+1}/"
+                f"{len(existing_variables)})..."
+            )
+            for month in range(1, 13):
+                logger.info(
+                    f"\t\tProcessing month {month!r} ({month}/12)...")
+                existing_measurements = database.collect_all_monthly_measurements(
+                    db_session,
+                    station_id_filter=station.id,
+                    variable_id_filter=variable.id,
+                    month_filter=month
+                )
+                existing = {}
+                for db_measurement in existing_measurements:
+                    measurement_id = _build_measurement_id(db_measurement)
+                    existing[measurement_id] = db_measurement
+                response = client.get(
+                    "https://api.arpa.veneto.it/REST/v1/provaclima/solo",
+                    params={
+                        "statcd": station.code,
+                        "indicatore": variable.name,
+                        "pdati": 0.95,
```

Comment:
Yes, I am aware of this and have opened #47 to track its implementation. 

I was done with this PR before the internal REST API got modified and have since moved on to tackle other issues - so it would be easier for me to cater to the internal REST API changes in a subsequent PR.